src/ApplicationDefaultCredentials.php

```diff
@@ -321,6 +328,40 @@ public static function getIdTokenCredentials(
         return $creds;
     }
 
+    /**
+     * Returns a StdOutLogger instance
+     *
+     * @return null|LoggerInterface
+     */
+    public static function getDefaultLogger(): null|LoggerInterface
+    {
+        $loggingFlag = getenv(self::SDK_DEBUG_FLAG);
+
+        // Env var is not set
+        if (!is_string($loggingFlag)) {
+            if (is_array($loggingFlag)) {
+                trigger_error('The ' . self::SDK_DEBUG_FLAG . ' is set, but it is set to another value than false, true, 0 or 1. Logging is disabled');
+                return null;
+            }
+
+            return null;
+        }
+
+        $loggingFlag = strtolower($loggingFlag);
+
+        // Env Var is not true
+        if ($loggingFlag !== 'true' && $loggingFlag !== '1') {
+            // Env var is set to a non valid value
+            if ($loggingFlag !== 'false' && $loggingFlag !== '0') {
+                trigger_error('The ' . self::SDK_DEBUG_FLAG . ' is set, but it is set to another value than false, true, 0 or 1. Logging is disabled');
+            }
+
+            return null;
+        }
+
+        return new StdOutLogger();
```

Comment:
Having lots of `!==` is a bit hard to read. What about something like this?

**NOTE**: You may want to handle the case of empty string (e.g. someone sets `GOOGLE_SDK_DEBUG_LOGGING=` in their environment) to have the same behavior, which is why I included the empty string in the array.

```suggestion
        // Env Var is true
        if ($loggingFlag === 'true' || $loggingFlag === '1') {
            return StdOutLogger();
        }

        // Env Var is false or not specified
        if (in_array($loggingFlag, [null, '', 'false', '0'], true)) {
            return null;
        }
        
        // Env var is set to a non valid value
        trigger_error('The ' . self::SDK_DEBUG_FLAG . ' is set, but it is set to another value than false, true, 0 or 1. Logging is disabled');

        // still return null because we don't want debugging to halt execution
        return null;
```

I personally think this reads a lot cleaner